packages/core/src/container/property-list/property-item-color.tsx

```diff
@@ -0,0 +1,90 @@
+import * as Components from '@meetalva/components';
+import * as MobxReact from 'mobx-react';
+import * as Model from '@meetalva/model';
+import * as React from 'react';
+import { ViewStore } from '../../store';
+import { debounce } from 'lodash';
+
+const OutsideClickHandler = require('react-outside-click-handler').default;
+
+export interface PropertyItemColorProps {
+	property: Model.ElementProperty;
+}
+
+export interface PropertyItemColorState {
+	propertyOverlay: boolean;
+}
+
+@MobxReact.inject('store')
+@MobxReact.observer
+export class PropertyItemColor extends React.Component<PropertyItemColorProps> {
+	public state = {
```

Comment:
I think we might get away without  internal state in `PropertyItemColor`.

1. `state.color` is a 1:1 copy of `props.property.getValue()`, so let's use that instead
2. `state.displayColorPicker`, `.showPropertyOverlay` and `.hidePropertyOverlay` should be moved to the `PatternProperty`

Reply:
Moving `tate.displayColorPicker` to the model would also simplify e.g. handling of clicks on the preview.

Reply:
A good candidate to handle `state.displayColorPicker` might be a `.focused` state on all element properties (not implemented yet). This could control the focus styling on "normal" input fields and the color picker toggling for this.